asmcomp/emitaux.ml

```diff
@@ -238,24 +238,66 @@ let emit_frames a =
     a.efa_def_label lbl;
     a.efa_string name
   in
-  let emit_defname (_filename, defname) (file_lbl, lbl) =
+  let emit_defname (_filename, defname, loc) (file_lbl, lbl) =
+    let emit_loc (start_chr, end_chr, end_offset) =
+      a.efa_16 start_chr;
+      a.efa_16 end_chr;
+      a.efa_32 (Int32.of_int end_offset)
+    in
     (* These must be 32-bit aligned, both because they contain a
        32-bit value, and because emit_debuginfo assumes the low 2 bits
        of their addresses are 0. *)
     a.efa_align 4;
     a.efa_def_label lbl;
     a.efa_label_rel file_lbl 0l;
+    (* Include the additional 64-bits of location information which didn't pack
+       in the main 64-bit word *)
+    Option.iter emit_loc loc;
     a.efa_string defname
   in
-  let pack_info fd_raise d has_next =
-    let line = Int.min 0xFFFFF d.Debuginfo.dinfo_line
-    and char_start = Int.min 0xFF d.Debuginfo.dinfo_char_start
-    and char_end = Int.min 0x3FF d.Debuginfo.dinfo_char_end
+  let fully_pack_info fd_raise d has_next =
+    (* Fully packed debuginfo:
+       0lllllmaaabbboooooddddddddddddkn
```

Comment:
I suggest this comment and the one at line 285ff should match the one in backtrace_nat.c.

Reply:
Actually, maybe it's better just to cross-reference it (as the out-of-sync-ness of it already demonstrates)?

Reply:
Just think, if we had a design document .... I agree, though. Put the information in one place, refer to it from the other, and have a back link so that when we move it we can fix the reference.